src/ctx.rs

```diff
@@ -121,12 +123,6 @@ impl Default for StrCtx {
     }
 }
 
-impl From<u8> for StrCtx {
```

Comment:
The type annotation may be even longer than a enum variance. :-/ It's really tricky to distinguish behaviour by integer type. If you really want to read fixed-length str by simply usize, you can impl `TryFromCtx<'a, &[u8], usize>` for str, which can coexist with the StrCtx one.